client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -92,6 +100,25 @@ public static String appUniqueId(SparkContext context) {
         .getOrElse(context::applicationId);
   }
 
+  public static int celebornShuffleId(
+      ShuffleClient client,
+      CelebornShuffleHandle<?, ?, ?> handle,
+      TaskContext context,
+      Boolean isWriter) {
+    if (handle.throwsFetchFailure()) {
+      int appShuffleId = handle.shuffleId();
+      int maxAttemptNum = handle.maxStageAttemptsNum();
+      int attemptId = context.stageAttemptNumber();
+      return appShuffleId * maxAttemptNum + attemptId;
```

Comment:
If this method is trying to create a new celeborn shuffle id always, why not simply use an `AtomicInteger` to generate a new id each time ?

Reply:
In current implementation, SparkUtils.celebornShuffleId(...) is called in shuffleWriter that is located in Spark executor, then there is no singleton AtomicInteger as a Id generator across the executors
We can use a single Id generator unless the id generation logic moves to Lifecyclemanger

Reply:
If this is in executor, how are we ensuring shuffle is = 0, stage 1.0 and shuffle is = 0, stage 3.0 (2.0 computing for same shuffle is due to fetch failures) have unique id ?

Reply:
this issue had been handled with new solution as described in https://github.com/apache/incubator-celeborn/pull/1924#discussion_r1381725261